services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3720,19 +3720,22 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
   @Override
   public void setStoreCurrentVersion(String clusterName, String storeName, int versionNumber) {
     storeMetadataUpdate(clusterName, storeName, store -> {
-      if (store.getCurrentVersion() != Store.NON_EXISTING_VERSION) {
-        if (versionNumber != Store.NON_EXISTING_VERSION && !store.containsVersion(versionNumber)) {
-          throw new VeniceException("Version:" + versionNumber + " does not exist for store:" + storeName);
-        }
+      // Parent colo should not update the current version of a store
+      if (!isParent()) {
```

Comment:
Can we do this check outside and avoid extra indentation...
```java
if (isParent()) {
  return;
}
```

Reply:
Sure